### DISCUSSION TITLE
Create PR?

### DISCUSSION THREAD

**Discussion by 6pac** [dotnet/aspnetcore#56280] [Q&A] [answered]
This may be an ultra-stupid question, but here goes.  FYI I am familiar with GitHub in general, run an open source project, but I'm used to people contributing to *my* repo, not the other way around!  
Have posted on Issue #6836, have cloned, branched and modded the source to allow a solution and would like to create a PR. Usually the way I do it is by pushing the branch and then going to GitHub to convert the branch to a PR. I don't appear to have permission. What do I need to do? This is, of course, if you actually want my PR =) 

    $  git push --set-upstream origin add-reponsecaching-policy-update
    ERROR: Permission to dotnet/aspnetcore.git denied to 6pac.
    fatal: Could not read from remote repository.
    
    Please make sure you have the correct access rights
    and the repository exists.

**Comment by En3Tho** [comment]
@6pac You seem to be pushing to origin. Can you check what origin is in your case? `git remote -v` can help you.

 There is a high possibly you're pushing into dotnet repo instead of your forked one. There is an error message indicating that.

**Reply by 6pac** [reply]
So this is the workfow so far:  

    git clone git@github.com:dotnet/aspnetcore.git --depth 1
    git checkout -b add-reponsecaching-policy-update
    (make changes to file)
    git add .
    git commit -m "commit message"
    git push --set-upstream origin add-reponsecaching-policy-update

git remote yields:

    $ git remote -v
    origin  git@github.com:dotnet/aspnetcore.git (fetch)
    origin  git@github.com:dotnet/aspnetcore.git (push)

As mentioned, I usually push a branch then in the GitHub GUI this can be converted to a PR.

**Reply by En3Tho** [reply, marked as answer]
@6pac Yeah the problem is that you have to first fork repo and either clone your fork (git will automatically set up your fork as origin) or add your fork using `git remote add ...`. Then you push to your fork and then create a pr. Generally your workflow is right except for not forking repo

**Reply by 6pac** [reply]
Thanks, a bit of a newb question, but probably good to have it recorded here.
